File: Sizing/LLC Sizing.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

// #ifndef pi
// #define pi 3.14159265358979323846
// #endif


int main() {
    // Dichiarazione delle variabili di input
    double Vin_min, Vin_max, Vin_nom, Vout, Pout, fr, M_nom, Qmax, m, Fxmin, Kmax;
    
    // Input dei valori
    printf("Inserisci Vin_min (Volt): ");
    scanf("%lf", &Vin_min);
    printf("Inserisci Vin_max (Volt): ");
    scanf("%lf", &Vin_max);
    printf("Inserisci Vin_nom (Volt): ");
    scanf("%lf", &Vin_nom);
    printf("Inserisci Vout (Volt): ");
    scanf("%lf", &Vout);
    printf("Inserisci Pout (Watt): ");
    scanf("%lf", &Pout);
    printf("Inserisci fr (Hertz): ");
    scanf("%lf", &fr);
    printf("Inserisci M_nom (1 = Full bridge, 0.5 = Half bridge): ");
    scanf("%lf", &M_nom);
    printf("Inserisci Qmax: ");
    scanf("%lf", &Qmax);
    printf("Inserisci m: ");
    scanf("%lf", &m);
    printf("Inserisci Fxmin: ");
    scanf("%lf", &Fxmin);
    printf("Inserisci Kmax: ");
    scanf("%lf", &Kmax);
    
    // Calcoli
    double pi = M_PI;
    double Np_Ns = (Vin_nom/Vout)*M_nom;
    double M_max = (Vin_nom/Vin_min)*M_nom;
    double M_min = (Vin_nom/Vin_max)*M_nom;
    double fs_min = Fxmin*fr;
    double Qmax_vmin = Qmax*(Vin_min/Vin_max);

    // K > M

    double Rac_min = (8/(pow(pi, 2)))*pow(Np_Ns, 2)*(pow(Vout, 2)/Pout);
    double Cr = sqrt(1/(pow(2*pi*fr, 2)*pow(Qmax*Rac_min, 2)));
    double Lr = pow(Qmax*Rac_min, 2)*Cr;
    double Lm = (m*Lr)-Lr;

    
    // Output dei risultati
    printf("\nOutput:\n");
    printf("Np/Ns: %lf\n", Np_Ns);
    printf("M_max: %lf\n", M_max);
    printf("M_min: %lf\n", M_min);
    printf("fs_min (Hertz): %lf\n", fs_min);
    printf("Qmax@vmin: %lf\n", Qmax_vmin);

    printf("Kmax: %lf e M_max: %lf\n", Kmax, M_max);

    printf("Rac_min (Ohm): %lf\n", Rac_min);
    printf("Lr (Henry): %0.15lf\n", Lr);
    printf("Cr (Farad): %0.15lf\n", Cr);
    printf("Lm (Henry): %0.15lf\n", Lm);

    //conversione
    double Lr_nH = Lr*1e9;
    double Cr_nF = Cr*1e9;
    double Lm_nH = Lm*1e9;

    printf("\nConversione:\n");
    printf("Lr (nH): %lf\n", Lr_nH);
    printf("Cr (nF): %lf\n", Cr_nF);
    printf("Lm (nH): %lf\n", Lm_nH);

    return 0;
}

File: Sizing/LLC Sizing.m
clc;
clear all;
close all;

% Dichiarazione delle variabili di input
Vin_min = input('Inserisci Vin_min (Volt): ');
Vin_max = input('Inserisci Vin_max (Volt): ');
Vin_nom = input('Inserisci Vin_nom (Volt): ');
Vout = input('Inserisci Vout (Volt): ');
Pout = input('Inserisci Pout (Watt): ');
fr = input('Inserisci fr (Hertz): ');
M_nom = input('Inserisci M_nom (1 = Full bridge, 0.5 = Half bridge): ');
Qmax = input('Inserisci Qmax: ');
m = input('Inserisci m: ');
Fxmin = input('Inserisci Fxmin: ');
Kmax = input('Inserisci Kmax: ');

% Calcoli
pi_value = pi; % Ottieni il valore di pi da MATLAB
Np_Ns = (Vin_nom/Vout)*M_nom;
M_max = double((Vin_nom/Vin_min)*M_nom);
M_min = double((Vin_nom/Vin_max)*M_nom);
fs_min = Fxmin*fr;
Qmax_vmin = Qmax*(Vin_min/Vin_max);

% Calcolo di Rac_min con valori numerici
Rac_min = double((8/(pi_value^2))*(Np_Ns^2)*((Vout^2)/Pout));

Cr = double(sqrt(1/(power(2*pi_value*fr, 2)*power(Qmax*Rac_min, 2))));
Lr = double(power(Qmax*Rac_min, 2)*Cr);
Lm = double((m*Lr)-Lr);

% Output dei risultati
fprintf('\nOutput:\n');
fprintf('Np/Ns: %f\n', Np_Ns);
fprintf('M_max: %f\n', M_max);
fprintf('M_min: %f\n', M_min);
fprintf('fs_min (Hertz): %f\n', fs_min);
fprintf('Qmax@vmin: %f\n', Qmax_vmin);

fprintf('Kmax: %f e M_max: %f\n', Kmax, M_max);

fprintf('Rac_min (Ohm): %f\n', Rac_min);
fprintf('Lr (Henry): %.15f\n', Lr);
fprintf('Cr (Farad): %.15f\n', Cr);
fprintf('Lm (Henry): %.15f\n', Lm);

% Conversione
Lr_nH = Lr*1e9;
Cr_nF = Cr*1e9;
Lm_nH = Lm*1e9;

fprintf('\nConversione:\n');
fprintf('Lr (nH): %f\n', Lr_nH);
fprintf('Cr (nF): %f\n', Cr_nF);
fprintf('Lm (nH): %f\n', Lm_nH);
